### PR TITLE
feat(ssrf): create multiple ssrf endpoints 

### PR DESCRIPTION
**Pull request by tamirGer**
I've added the `/api/file/google`, `/api/file/aws`, `/api/file/azure` and `/api/file/digital_ocean` endpoints which only allow ssrf payloads for the respective provider.

### CODE REVIEW

**Comment by bararchy** [comment]
@tamirGer can you also update the README file with this new change? (the SSRF part)

**Comment by tamirGer** [comment]
> @tamirGer can you also update the README file with this new change? (the SSRF part)

Done :)

File: src/file/file.controller.ts [hunk 1]
```diff
@@ -121,7 +251,7 @@ export class FileController {
     description: SWAGGER_DESC_READ_FILE_ON_SERVER,
   })
   @ApiNotFoundResponse({
-    description: 'File not found',
+    description: 'File not Found',
```
Comment (denis-maiorov-brightsec) [inline src/file/file.controller.ts:254]: this change is not related to `multiple ssrf endpoints`, please revert

File: src/file/file.controller.ts [hunk 2]
```diff
@@ -108,11 +238,11 @@ export class FileController {
       if (typeof raw === 'string' || Buffer.isBuffer(raw)) {
         await fs.promises.access(path.dirname(file), W_OK);
         await fs.promises.writeFile(file, raw);
-        return `File uploaded successfully at ${file}`;
+        return 'File uploaded successfully';
```
Comment (denis-maiorov-brightsec) [inline src/file/file.controller.ts:241]: this change is not related to `multiple ssrf endpoints`, please revert

File: src/file/file.controller.ts [hunk 3]
```diff
@@ -37,6 +39,26 @@ export class FileController {
 
   constructor(private fileService: FileService) {}
 
+  private getContentType = (contentType: string, acceptHeader: string) => {
```
Comment (denis-maiorov-brightsec) [inline src/file/file.controller.ts:42]: what's the purpose of arrow function here? shouldn't it be just private method for the sake of consistency?

File: src/file/file.controller.ts [hunk 4]
```diff
@@ -37,6 +39,26 @@ export class FileController {
...
+      return contentType;
+    } else if (acceptHeader) {
+      return acceptHeader;
+    } else {
+      return 'application/octet-stream';
+    }
+  }
+
+  private async loadCPFile(cp_name: string, path: string){
+    if (!path.startsWith(CloudProvidersMetaData[cp_name.toUpperCase()])){
```
Comment (denis-maiorov-brightsec) [inline src/file/file.controller.ts:53]:
```suggestion
    if (!path.startsWith(cpBaseUrl)){
```

File: src/file/file.controller.ts [hunk 5]
```diff
@@ -59,17 +81,125 @@ export class FileController {
...
+  @ApiOperation({
+    description: SWAGGER_DESC_READ_FILE,
+  })
+  async loadDigitalOceanFile(
+    @Query('path') path: string,
+    @Query('type') contentType: string,
+    @Res({ passthrough: true }) res: FastifyReply,
+    @Headers('accept') acceptHeader: string,
+  ) {
+    const file: Stream = await this.loadCPFile('DIGITAL_OCEAN', path);
```
Comment (denis-maiorov-brightsec) [inline src/file/file.controller.ts:201]:
```suggestion
    const file: Stream = await this.loadCPFile(CloudProvidersMetaData.DIGITAL_OCEAN, path);
```

**Review by denis-maiorov-brightsec — CHANGES_REQUESTED** [review]

File: src/file/file.controller.ts (hunk 3 above)
Comment (tamirGer) [reply src/file/file.controller.ts]: You're right, updating :)

**Comment by bararchy** [comment]
@tamirGer when everything is done, please re-assign Denis to review

**Review by denis-maiorov-brightsec — APPROVED** [review]
Looks good to me

**Comment by bararchy** [comment]
@denis-maiorov-brightsec I think those should have went to `master` and not `development`

**Comment by denis-maiorov-brightsec** [comment]
@bararchy why do you think it should go to `master`? All other PRs are merged to `development` as I see

**Comment by bararchy** [comment]
@denis-maiorov-brightsec not true, the latest change, which is the search change, got pushed into master, and it's deployed. I'm not sure where the `development` branch is going to

**Comment by Kif-Fando** [comment]
@bararchy @denis-maiorov-brightsec are right, usually all PRs goes to development and after some time they go to master. Latest PR went directly to master i don't know why. My latest PR was merged to development, but it still there and didn't merged to master. All previous PRs were merged by Artem.
